perf(scope): speed up dependency lookup with Map-based binding resolv…

pese-git opened the pull request:
…er index

Optimize resolve()/tryResolve() to use O(1) Map-based lookup by type and name instead of iterating through all modules and bindings. Behavior of factory, singleton, instance, and named bindings is preserved.

pese-git committed:
perf(scope): speed up dependency lookup with Map-based binding resolver index

Optimize resolve()/tryResolve() to use O(1) Map-based lookup by type and name instead of iterating through all modules and bindings. Behavior of factory, singleton, instance, and named bindings is preserved.

pese-git committed:
docs: update EN/RU quick start and tutorial with Fast Map-based lookup section; clarify performance benefit in README

pese-git committed:
docs(perf): clarify Map-based resolver optimization applies since v3.0.0 in all docs

KlimYarosh commented on cherrypick/lib/src/scope.dart:
```diff
@@ -254,16 +259,20 @@ class Scope with CycleDetectionMixin, GlobalCycleDetectionMixin {
   }
 
   BindingResolver<T>? _findBindingResolver<T>(String? named) {
+    final byType = _bindingResolvers[T];
+    if (byType == null) return null;
+    return byType[named] as BindingResolver<T>?;
```
```suggestion
    return _bindingResolvers[T]?[named] as BindingResolver<T>?;
```

pese-git committed:
refactor(scope): simplify _findBindingResolver<T> with one-liner and optional chaining

The function is now shorter, more readable and uses modern Dart null-safety idioms. No functional change.
